audio.c

```diff
@@ -88,98 +88,26 @@ writer(void *priv)
 	}
 
 	if (mp3) {
-		lame = lame_init();
-		if (lame == NULL) {
-			fprintf(stderr, "Couldn't initialize lame encoder\n");
-			exit(EXIT_FAILURE);
-		}
-
-		buf_size = (1.25 * BUF_TIME_S * ctx.stream->sample_rate) + 7200;
-		mp3buf = malloc(buf_size);
-		if (mp3buf == NULL) {
-			fprintf(stderr, "No memory for mp3 encoding buffer\n");
-			exit(EXIT_FAILURE);
-		}
-
-		lame_set_num_channels(lame, ctx.stream->layout.channel_count);
-		lame_set_mode(lame, ctx.mono ? MONO : STEREO);
-		lame_set_error_protection(lame, 1);
-		lame_set_in_samplerate(lame, ctx.stream->sample_rate);
-		lame_set_findReplayGain(lame, 1);
-		lame_set_asm_optimizations(lame, MMX, 1);
-		lame_set_asm_optimizations(lame, SSE, 1);
-		lame_set_quality(lame, 3);
-		lame_set_bWriteVbrTag(lame, 1);
-		lame_set_VBR(lame, vbr_mtrh);
-		lame_set_VBR_q(lame, 3);
-		lame_set_VBR_min_bitrate_kbps(lame, 96);
-		lame_set_VBR_max_bitrate_kbps(lame, 320);
+		aw = audio_writer_lame(ctx.fout, ctx.stream->sample_rate,
+					   ctx.stream->layout.channel_count,
+					   BUF_TIME_S, ctx.mono);
```

Comment:
I prefer using something stylistically similar to FreeBSD's style(9). There is some deviance in my interpretation (in particular, all control flow structures must use blocks, maximum line width is a little fluid around 100 characters, not 80). In this style, statements continue to the next line with four spaces. So for this code, the suggestion would be e.g.

```c
        aw = audio_writer_lame(ctx.fout, ctx.stream->sample_rate,
            ctx.stream->layout.channel_count, BUF_TIME_S, ctx.mono);
```